server.js

```diff
@@ -1,38 +1,140 @@
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+
+import bookData from './data/books.json';
+
+// Allowing for us to connect to mongodb database
+// Line 10 url to database which we can use in mongodb to access the server data
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/books";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Creating the model/structure of each of the instances that will be entered into the database. This model is based on the object structure/keys/values defined for the array elements in the books.json data set. We write the key name and the value type of each of the object keys as defined in the books.json.
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: String,
+  isbn13: String,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number,
+});
+
+/* Seeding database: 
+1. Seeding database means that the data defined in the function is sent to our database when the database is first started.
+2. Mongoose async/await operations is being used to wait for the operation deleteMany to finish before it goes on to re-populate the database with the array of books. 
+3. Creating a new instance based on the Books mongoose model above and the books.json array of data which is being iterated over and saved to the mongoDB database.
+4. If statement will be true if I have entered the environment variable RESET_DB and equalled it to true when doing npm run dev in the terminal. This will signify that I want the database to be re-run and that the seeding of the database will begin!
+*/
+if(process.env.RESET_DB) {
+  const seedDatabase = async () => {
+  // deleteMany will allow for the data that was stored on the database on the last get request to be deleted so that there are no dupications of data. Then the new Authors will be created again.
+    await Book.deleteMany({});
+  
+    bookData.forEach((bookData) => {
+      new Book(bookData).save();
+    })
+  };
+  
+  seedDatabase();
+}
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+// First endpoint that will return the whole array of books data
+// Async and await are being used as find is asyncronous and might take some time to return the data being fetched from the server.
+app.get('/books', async (request, response) => {
```

Comment:
Is find() as a method asynchronous that needs await/async as you say in your comment, or is a http request asynchronous? And we handle that with await/async?